### PR TITLE
Refactor project setup + update to RN 68.2 + update dev dependencies

### PR DESCRIPTION
**Pull request by AlexanderEggers**
This PR bases on https://github.com/lottie-react-native/lottie-react-native/pull/877 but includes a decent amount of additional changes to make the project better aligned to the rest of open source libraries out there. These changes are needed so that I can have two fully different example projects: one for fabric and one without fabric.

### Summary of the major setup changes
- The new setup creates two fully separated areas: lib (root) and example
- The lib is the root folder (as before) which takes care of actually containing the lib source code and building the lib files via babel.
- The example is fully separated from the lib and only has a "weak" reference to that one via `"lottie-react-native": "file:.."`. That allows us to build multiple example projects without having to deal with the additional complexity in the root folder which should just take care of the lib creation.
    - Because of these changes I was able to simplify the examples quite a lot and make them a lot closer to how this library will be used in a final project.
    - Android can finally use `npx react-native ...` to be compiled instead of the strange previous way.
- Remove npm offline usage. That is causing some problems with the new setup.
 
### Testing via example project
- [x] Android
- [x] iOS
- [x] Windows - I need help with that. I don't have a Windows setup for this.

### Testing for other project related areas
- [x] Lib folder creation with Babel - The lib output files look fine to me and seem to work when using them via the example project.
- [ ] NPM deployment - @emilioicai I think that only you who can help me verify that part.

### Additionally this PR includes some project specific changes
- Using the RN 0.68.2 [upgrade helper ](https://react-native-community.github.io/upgrade-helper/)to make changes to the whole project.
- Update dev dependencies - there were plenty which were extremely outdated and I have not noticed any issues so far. Please correct me if I missed something. I am happy to revert some of these dependency updates.

### Next
With having a better project setup, I would start adding the actual RN new architecture code as part of the next PR. It would be great if we could have this one merged in the next days so that I can get really started on the fabric part.

### CODE REVIEW

**Comment by Willham12** [comment]
@AlexanderEggers thanks for the contribution, are there any news on this PR?

**Comment by AlexanderEggers** [comment]
@Willham12 I am still waiting for feedback by @emilioicai  for this PR and I need help to verify if the publishing and thd windows version are still working fine.

**Comment by AlexanderEggers** [comment]
@emilioicai Can we merge this PR prior to any additional changes to develop? Otherwise the changes are going to be extremely difficult for me to maintain. I noticed that you merged some changes which require a rebase from my side now. I will try to take care of that some time tomorrow.

**Comment by emilioicai** [comment]
Hey Alex, yes, sorry for not being  that active, these weeks have been quite hectic for me. I started reviewing and hopefully merge before the week ends. I promise I won't commit any large changes into the repo before this PR is merged

**Comment by matinzd** [comment]
It seems that there are some merge conflicts @AlexanderEggers. Can you fix them?

**Comment by matinzd** [comment]
Isn’t it better to upgrade to 0.69.3. It has been a while that this PR is still open.

**Comment by AlexanderEggers** [comment]
> 

At some stage, yes. But I want to avoid to keep updating this PR and just have a base ready for now.

File: lottie-react-native.podspec
Comment (matinzd) [inline lottie-react-native.podspec:15]: This needs to be updated to: `https://github.com/lottie-react-native/lottie-react-native.git`

**Review by matinzd — CHANGES_REQUESTED** [review]
Thank you for your pull request and this is a huge step for the future of the repo. So happy to see `those npm-packages-offline-cache` are all removed.
It seems like example app for both Android and iOS is not working after fixing the build myself. 
```
 ERROR  Invariant Violation: View config getter callback for component `LottieAnimationView` must be a function (received `undefined`).

This error is located at:
    in LottieAnimationView (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at LottieView.js:178)
    in RCTView (at View.js:32)
    in View (at LottieView.js:177)
    in LottieView (at LottieAnimatedExample.js:170)
    in RCTView (at View.js:32)
    in View (at LottieAnimatedExample.js:169)
    in RCTView (at View.js:32)
    in View (at LottieAnimatedExample.js:163)
    in LottieAnimatedExample (at renderApplication.js:50)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:92)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:43)
    in example(RootComponent) (at renderApplication.js:60)
```

I also added some other comments up there. 

**UPDATE:** 
I found the issue with the example app. Module's`node_modules` folder should not get included inside example's `node_modules` causing confusion between peer dependencies. After removing inner `node_modules` folder app started working both on android and iOS. 
It seems like metro config should be fixed to blacklist inner `node_modules` inside installed project.

- example
   - node_modules
      - lottie-react-native
          - node_modules <-- this should be ignored

File: example/metro.config.js
Comment (matinzd) [inline example/metro.config.js:17]: I would suggest this metro config for example app.

```suggestion
const path = require('path');
const exclusionList = require('metro-config/src/defaults/exclusionList');
const escape = require('escape-string-regexp');
const pack = require('../package.json');

const root = path.resolve(__dirname, '..');

const modules = [...Object.keys(pack.peerDependencies)];

module.exports = {
  projectRoot: __dirname,
  watchFolders: [root],

  // We need to make sure that only one version is loaded for peerDependencies
  // So we exclude them at the root, and alias them to the versions in example's node_modules
  resolver: {
    blacklistRE: exclusionList(
      modules.map(
        m => new RegExp(`^${escape(path.join(root, 'node_modules', m))}\\/.*$`),
      ),
    ),

    extraNodeModules: modules.reduce((acc, name) => {
      acc[name] = path.join(__dirname, 'node_modules', name);
      return acc;
    }, {}),
  },

  transformer: {
    getTransformOptions: async () => ({
      transform: {
        experimentalImportSupport: false,
        inlineRequires: true,
      },
    }),
  },
};
```

**Comment by matinzd** [comment]
You can just pick this commit if you don't want to put time into fixing the comments. 
https://github.com/lottie-react-native/lottie-react-native/commit/aa10ae3040f7a02f1a6fcfa7a98e53962c553403

**Comment by AlexanderEggers** [comment]
@matinzd Thanks for doing the review. I will try to have a look at your comments by the end of this week.

**Comment by AlexanderEggers** [comment]
@matinzd Thanks again for the review. I cherry-picked your commit into this branch and added a few minor additional changes. Let me know if you can notice anything else.

**Comment by matinzd** [comment]
> @matinzd Thanks again for the review. I cherry-picked your commit into this branch and added a few minor additional changes. Let me know if you can notice anything else.

I am going to test windows today on my wife’s latpop and setup the environment 😅.

**Comment by AlexanderEggers** [comment]
@emilioicai @matinzd I started working on the fabric support. I think it would be great if we can get this first part either merged into `/master` or we create a `/fabric` branch which we will use internally for now until it is stable enough for the public. What do you think?

The relevant work is currently sitting in this branch: https://github.com/AlexanderEggers/lottie-react-native/tree/feature/add-fabric-support I haven't had a chance to work much on that. I just migrated the android and JS code to technically support fabric but I am having troubles running the non-fabric example project at the moment.

Let me know how you want to proceed with this PR and the future fabric work.

**Comment by matinzd** [comment]
> @emilioicai @matinzd I started working on the fabric support. I think it would be great if we can get this first part either merged into `/master` or we create a `/fabric` branch which we will use internally for now until it is stable enough for the public. What do you think?
> 
> The relevant work is currently sitting in this branch: https://github.com/AlexanderEggers/lottie-react-native/tree/feature/add-fabric-support I haven't had a chance to work much on that. I just migrated the android and JS code to technically support fabric but I am having troubles running the non-fabric example project at the moment.
> 
> Let me know how you want to proceed with this PR and the future fabric work.

That would be great! We also need to start working on fabric and make it backward compatible. There is a repo in react native community with examples for making fabric backward compatible, so we can use that. 

https://github.com/cipolleschi/RNNewArchitectureLibraries/tree/feat/back-fabric-component-070

**Comment by AlexanderEggers** [comment]
@matinzd Yep I'm using that one 👍 Can you create a `/fabric` branch based on master so that I can retarget this PR?

**Comment by emilioicai** [comment]
@matinzd I finally managed to have the windows support tested and everything looks good on that front 🙌 
@AlexanderEggers I created that [fabric branch](https://github.com/lottie-react-native/lottie-react-native/tree/fabric) for you to target, thanks again for your support!

**Comment by AlexanderEggers** [comment]
@emilioicai Thanks for creating the branch. I re-targeted the branch to that branch now. I think the only remaining thing we need to verify is the NPM deployment but we could also do that when we closer to the actual going public, what do you think?

**Comment by matinzd** [comment]
@emilioicai 
Where are we going to release v6 if we are targeting this on fabric branch?

**Review by matinzd — APPROVED** [review]

**Comment by AlexanderEggers** [comment]
> About NPM deployment we can start with `lottie-react-native@next` for /fabric branch?

I think that's a great idea 👍

**Comment by matinzd** [comment]
Let's change base branch to master again so that @emilioicai can release the latest changes as a patch. Then we can start `fabric` branch based on master. @AlexanderEggers

**Comment by AlexanderEggers** [comment]
@emilioicai Would it be possible if we can get this PR merged into `/fabric` this weekend? I'm pretty close with the android fabric work and got actually plenty of notes that we need to discuss together.

**Review by emilioicai — APPROVED** [review]

**Pull request merged**
